drivers/sdl/joypad_sdl.cpp

```diff
@@ -0,0 +1,385 @@
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "joypad_sdl.h"
+
+#ifdef SDL_ENABLED
+
+#include "core/error/error_macros.h"
+#include <iterator>
+
+#include "SDL3/SDL.h"
+#include "SDL3/SDL_error.h"
+#include "SDL3/SDL_events.h"
+#include "SDL3/SDL_gamepad.h"
+#include "SDL3/SDL_iostream.h"
+#include "SDL3/SDL_joystick.h"
+
+#include "core/input/default_controller_mappings.h"
+#include "core/os/time.h"
+
+JoypadSDL *JoypadSDL::singleton = nullptr;
+
+void JoypadSDL::process_inputs_thread_func(void *p_userdata) {
+	JoypadSDL *joy = static_cast<JoypadSDL *>(p_userdata);
+	joy->process_inputs_run();
+}
+
+#define HANDLE_SDL_ERROR(call) \
+	error = call;              \
+	ERR_FAIL_COND_V_MSG(error != 0, FAILED, SDL_GetError())
+
+// Macro to skip the SDL joystick event handling if the device is an SDL gamepad, because
+// there are separate events for SDL gamepads
+#define SKIP_EVENT_FOR_GAMEPAD                                      \
+	if (SDL_GetGamepadFromID(sdl_event.jbutton.which) != nullptr) { \
+		continue;                                                   \
+	}
+
+#define SETUP_JOYPAD_EVENT(event_type)                                \
+	joypad_event.timestamp = Time::get_singleton()->get_ticks_usec(); \
+	joypad_event.type = JoypadEventType::event_type;                  \
```

Comment:
```suggestion
#define SETUP_JOYPAD_EVENT(m_event_type)                              \
	joypad_event.timestamp = Time::get_singleton()->get_ticks_usec(); \
	joypad_event.type = JoypadEventType::m_event_type;                \
```
Macro naming standards